frontend/public/components/ingress.jsx

```diff
@@ -97,14 +97,24 @@ const RulesRows = (props) => {
 
   if (_.has(props.spec, 'rules')) {
     _.forEach(props.spec.rules, rule => {
```

Comment:
Making sure I understand the logic here.  I'm not super familiar with `Ingress`. 

`ingres.spec` should have `rules`.  If it doesn't, we will render `<div className="co-m-table-grid__body">`, but it will be empty.

Each rule should have an `http.paths` prop, which may have multiple values.  If so, in branch 2 we just flatten these out for easier rendering with `<RulesRow>`.

If it does not have `http.paths`, we can create a stand-in rule with the `ingress.spec.backend.*` data.  This may or may not exist.  

Is the `Ingress` in an ok state if we trigger `_.isEmpty(paths)` and *possibly* create a rule to render on the fly?  

Trying to decide if there could be a named helper in here or not, just due to the depth of if/loop/if/loop.

Reply:
> Is the Ingress in an ok state if we trigger _.isEmpty(paths) and possibly create a rule to render on the fly?

I'm not clear on what this means. (But what's above it looks right.)

Reply:
There is a rule, it just doesn't have an http.paths property, which is optional.

https://github.com/kubernetes/api/blob/d77153c7cc7a359e9edf2fa7afdc0f6303240382/extensions/v1beta1/types.go#L649-L674

```go
	// IngressRuleValue represents a rule to route requests for this IngressRule.
	// If unspecified, the rule defaults to a http catch-all. Whether that sends
	// just traffic matching the host to the default backend or all traffic to the
	// default backend, is left to the controller fulfilling the Ingress. Http is
	// currently the only supported IngressRuleValue.
```

Reply:
Ok great! Think I'm good.

/lgtm